pkg/manager/cli_download_install.go

```diff
@@ -0,0 +1,319 @@
+		}
+	}
+
+	if len(containerEnvVars) > 0 {
+		dep.Spec.Template.Spec.Containers[0].Env = containerEnvVars
+	}
+
+	return dep, nil
+}
+
+func getService(log logr.Logger) (*corev1.Service, error) {
+	serviceFile, err := fs.ReadFile("manifests/cli/service.yaml")
+	if err != nil {
+		log.Error(err, "failed to read manifests/cli/service.yaml")
+		return nil, err
+	}
+
+	service := &corev1.Service{}
+	err = yaml.Unmarshal(serviceFile, &service)
+	if err != nil {
+		log.Error(err, "failed to parse manifests/cli/service.yaml")
+		return nil, err
+	}
+
+	return service, nil
+}
+
+func getRoute(log logr.Logger) (*routev1.Route, error) {
+	routeFile, err := fs.ReadFile("manifests/cli/route.yaml")
+	if err != nil {
+		log.Error(err, "failed to read manifests/cli/route.yaml")
+		return nil, err
+	}
+
+	route := &routev1.Route{}
+	err = yaml.Unmarshal(routeFile, &route)
+	if err != nil {
+		log.Error(err, "failed to parse manifests/cli/route.yaml")
+		return nil, err
+	}
+
+	return route, nil
+}
+
+func getConsoleDownload(routeUrl string, log logr.Logger) (*consolev1.ConsoleCLIDownload, error) {
+	log.Info("using route URL: " + routeUrl)
+	cliDownloadFile, err := fs.ReadFile("manifests/cli/consoledownload.yaml")
+	if err != nil {
+		log.Error(err, "failed to read manifests/cli/consoledownload.yaml")
+		return nil, err
+	}
+
+	cliDownload := &consolev1.ConsoleCLIDownload{}
+	err = yaml.Unmarshal(cliDownloadFile, &cliDownload)
+	if err != nil {
+		log.Error(err, "failed to parse manifests/cli/consoledownload.yaml")
+		return nil, err
+	}
+
+	links := []consolev1.CLIDownloadLink{
+		{
+			Href: "https://" + routeUrl + "/hypershift-linux-amd64.tar.gz",
+			Text: "Download oc hcp for Linux for x86_64",
+		},
+		{
+			Href: "https://" + routeUrl + "/hypershift-darwin-amd64.tar.gz",
+			Text: "Download oc hcp for Mac for x86_64",
+		},
+		{
+			Href: "https://" + routeUrl + "/hypershift-windows-amd64.tar.gz",
+			Text: "Download oc hcp for Windows for x86_64",
+		},
+		{
+			Href: "https://" + routeUrl + "/hypershift-linux-arm64.tar.gz",
+			Text: "Download oc hcp for Linux for ARM 64",
+		},
+		{
+			Href: "https://" + routeUrl + "/hypershift-darwin-arm64.tar.gz",
+			Text: "Download oc hcp for Mac for ARM 64",
+		},
+	}
+	cliDownload.Spec.Links = links
+
+	return cliDownload, nil
+}
+
+func getOwnerRef(hubclient client.Client, log logr.Logger) (*metav1.OwnerReference, []corev1.EnvVar, error) {
+	deployment := &appsv1.Deployment{}
+	err := hubclient.Get(context.TODO(), types.NamespacedName{Namespace: "multicluster-engine", Name: "hypershift-addon-manager"}, deployment)
+	if err != nil {
+		log.Error(err, "failed to get hypershift-addon-manager deployment")
+		return nil, nil, err
+	}
+
+	ownerRef := metav1.NewControllerRef(deployment.GetObjectMeta(), schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "Deployment"})
+	return ownerRef, deployment.Spec.Template.Spec.Containers[0].Env, nil
```

Comment:
for `deployment.Spec.Template.Spec.Containers[0].Env`
Do we have to worry about nilpointer or indexoutofbound here? Or it will always be there so don't bother.

Reply:
I am going after a very specific container we know about https://github.com/stolostron/backplane-operator/blob/main/pkg/templates/charts/toggle/hypershift/templates/hypershift-addon-manager-deployment.yaml#L69